File: SDKs/amazonaws.com/eventbridge/2015-10-07/go/pkg/models/shared/replay.go
package shared

import (
	"time"
)

type Replay struct {
	EventEndTime          *time.Time       `json:"EventEndTime,omitempty"`
	EventLastReplayedTime *time.Time       `json:"EventLastReplayedTime,omitempty"`
	EventSourceArn        *string          `json:"EventSourceArn,omitempty"`
	EventStartTime        *time.Time       `json:"EventStartTime,omitempty"`
	ReplayEndTime         *time.Time       `json:"ReplayEndTime,omitempty"`
	ReplayName            *string          `json:"ReplayName,omitempty"`
	ReplayStartTime       *time.Time       `json:"ReplayStartTime,omitempty"`
	State                 *ReplayStateEnum `json:"State,omitempty"`
	StateReason           *string          `json:"StateReason,omitempty"`
}

File: SDKs/amazonaws.com/events/2015-10-07/typescript/src/sdk/models/shared/replay.ts
import { Metadata, SpeakeasyBase } from "../../../internal/utils/utils";
import { ReplayStateEnum } from "./replaystateenum";


// Replay
/** 
 * A <code>Replay</code> object that contains details about a replay.
**/
export class Replay extends SpeakeasyBase {
  @Metadata({ data: "json, name=EventEndTime" })
  eventEndTime?: Date;

  @Metadata({ data: "json, name=EventLastReplayedTime" })
  eventLastReplayedTime?: Date;

  @Metadata({ data: "json, name=EventSourceArn" })
  eventSourceArn?: string;

  @Metadata({ data: "json, name=EventStartTime" })
  eventStartTime?: Date;

  @Metadata({ data: "json, name=ReplayEndTime" })
  replayEndTime?: Date;

  @Metadata({ data: "json, name=ReplayName" })
  replayName?: string;

  @Metadata({ data: "json, name=ReplayStartTime" })
  replayStartTime?: Date;

  @Metadata({ data: "json, name=State" })
  state?: ReplayStateEnum;

  @Metadata({ data: "json, name=StateReason" })
  stateReason?: string;
}
